### PR TITLE
TR-2003 - Resolve issue with requests not working when subaccount ID is required

### PR DESCRIPTION
**Pull request by nicklemmon**
[TR-2003](https://jira.int.messagesystems.com/browse/TR-2003)

### What Changed
- Updated calls to `getDraft` and `getPublished` to pass template `subaccount_id`

### How To Test
1. Create a new template and assign it to a subaccount
2. On the templates list page, click on the duplicate button in both the recent activity section and the table
3. Verify that duplicating the template succeeds in both instances

### To Do
- [x] Incorporate feedback

### CODE REVIEW

File: src/pages/templates/components/settings/SettingsForm.js [hunk 1]
```diff
@@ -17,14 +17,14 @@ export default class SettingsForm extends React.Component {
...
-      subaccountId,
+      template,
       showAlert,
       content,
       setHasSaved
     } = this.props;
 
     const { subject: _subject, from: _from, reply_to: _reply_to, ...rest } = content; //content object has updated values for html, text etc, provided by useEditorContext
-
+    const subaccountId = template.subaccount_id;
```
Comment (bkemper) [inline src/pages/templates/components/settings/SettingsForm.js]: Use `draft`

File: src/pages/templates/components/editorActions/DuplicateTemplateModal.js [hunk 2]
```diff
@@ -69,10 +69,11 @@ const DuplicateTemplateModal = (props) => {
       createTemplate({
         name: draftName,
         id: draftId,
+        subaccount: template.subaccount_id,
+        sharedWithSubaccounts: template.shared_with_subaccounts,
         content: contentToDuplicate,
         parsedTestData: testDataToDuplicate,
-        options: template.options,
-        shared_with_subaccounts: template.shared_with_subaccounts
+        options: template.options
```
Comment (bkemper) [inline src/pages/templates/components/editorActions/DuplicateTemplateModal.js]: This need to use draft or published

File: src/pages/templates/components/settings/SettingsForm.js (hunk 1 above)
Comment (nicklemmon) [reply src/pages/templates/components/settings/SettingsForm.js]: Done.

File: src/pages/templates/components/editorActions/DuplicateTemplateModal.js (hunk 2 above)
Comment (nicklemmon) [reply src/pages/templates/components/editorActions/DuplicateTemplateModal.js]: Done.

**Pull request merged**